### PR TITLE
bpo-38157: Add example about per file output for mock_open.

### PR DESCRIPTION
**Pull request by tirkarthi**
* Add an example with using `side_effect` to return content per file.

https://bugs.python.org/issue38157

### CODE REVIEW

**Comment by matrixise** [comment]
@tirkarthi Do you want to add a blurb entry?

**Comment by tirkarthi** [comment]
I feel it's not necessary since it's adding only documentation example.

**Review by corona10 — APPROVED** [review]
@tirkarthi  
Looks good to me.
Let's wait for core developers review :)

Thanks for your contribution.

**Comment by cjw296** [comment]
Would it be possible to add a unit test that exercises this pattern too?
(I wish we could do literal includes with the cpython Sphinx docs, so the examples actually get tested, thoughts?)

**Comment by tirkarthi** [comment]
> Would it be possible to add a unit test that exercises this pattern too?

unit test as a test in code or as doctest itself like below?

```
>>> DEFAULT = "default"
>>> data_dict = {"file1": "data1", "file2": "data2"}
>>> def open_side_effect(name):
...     return mock_open(read_data=data_dict.get(name, DEFAULT))()
...
>>> with patch("builtins.open", side_effect=open_side_effect):
...     with open("file1") as file1:
...         assert file1.read() == "data1"
...     with open("file2") as file2:
...         assert file2.read() == "data2"
...     with open("file3") as file2:
...         assert file2.read() == "default"
```

**Comment by cjw296** [comment]
If we could do literal includes, then we could include the actual unit test that gets run. Not sure that's possible?

A doctest such as the one you show: where would that get executed (and so tested...)?

File: Doc/library/unittest.mock-examples.rst
Comment (slateny) [inline Doc/library/unittest.mock-examples.rst]:
```suggestion
contents per file stored in a dictionary::
```

File: Doc/library/unittest.mock-examples.rst
Comment (slateny) [inline Doc/library/unittest.mock-examples.rst]: This should probably be initially indented 3 spaces as convention (see the example `def f(a, b, c): pass` and [devguide](https://devguide.python.org/documentation/style-guide/#use-of-whitespace)) above.

**Review by hugovk — APPROVED** [review]
Thanks!

**Comment by hugovk** [comment]
@corona10:

> @tirkarthi
> Looks good to me.
> Let's wait for core developers review :)

You're both core devs now :) :tada:

**Comment by tirkarthi** [comment]
Thanks @hugovk

**Pull request merged**
